### PR TITLE
Effects made accessible in watchdog handlers

### PR DESCRIPTION
**Pull request by lhstrh**
The grammar allows effects to be specified for a watchdog, but the compiler wasn't resolving effects correctly and the code generator needed to be updated to ensure that effects are brought into scope. This PR also updates the diagram synthesis to render watchdog effects correctly.

### CODE REVIEW

File: core/src/main/java/org/lflang/generator/c/CWatchdogGenerator.java [hunk 1]
```diff
@@ -9,16 +9,12 @@
...
 import org.lflang.ast.ASTUtils;
 import org.lflang.generator.CodeBuilder;
 import org.lflang.generator.ReactorInstance;
-import org.lflang.lf.Mode;
-import org.lflang.lf.ModeTransition;
-import org.lflang.lf.Reactor;
-import org.lflang.lf.VarRef;
-import org.lflang.lf.Variable;
-import org.lflang.lf.Watchdog;
+import org.lflang.lf.*;
```
Comment (lhstrh) [inline core/src/main/java/org/lflang/generator/c/CWatchdogGenerator.java:17]: The style guide does not allow wildcard imports.

File: test/C/src/concurrent/WatchdogAction.lf [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+ * watchdog expirations. This version uses an action instead of a reaction to the watchdog.
+ * @author Benjamin Asch
+ * @author Edward A. Lee
+ */
+target C {
+  timeout: 11000 ms
+}
+
+reactor Watcher(timeout: time = 1500 ms) {
+  // Offset ameliorates startup time.
```
Comment (lhstrh) [inline test/C/src/concurrent/WatchdogAction.lf:13]: What startup time is this referring to?

File: core/src/main/java/org/lflang/generator/c/CWatchdogGenerator.java (hunk 1 above)
Comment (edwardalee) [reply core/src/main/java/org/lflang/generator/c/CWatchdogGenerator.java:17]: Intellij does this automatically.  Here are the places where this appears, most of which have nothing to do with this PR:

image

File: test/C/src/concurrent/WatchdogAction.lf (hunk 2 above)
Comment (edwardalee) [reply test/C/src/concurrent/WatchdogAction.lf:13]: The time it takes the program to start up.

**Comment by edwardalee** [comment]
I've pushed support here for the diagrams, so that when a watchdog declares an action as an effect, we get this:

image

File: test/C/src/concurrent/WatchdogAction.lf (hunk 2 above)
Comment (lhstrh) [reply test/C/src/concurrent/WatchdogAction.lf:13]: I think "ameliorates" is a bit odd in this sentence. Did you mean "amortizes"?

File: core/src/main/java/org/lflang/generator/c/CWatchdogGenerator.java (hunk 1 above)
Comment (lhstrh) [reply core/src/main/java/org/lflang/generator/c/CWatchdogGenerator.java:17]: It must have slipped past reviews before, but that doesn't change the fact that the style guide prohibits it... Doesn't IntelliJ ask for confirmation before it carries out such automated refactoring?

File: core/src/main/java/org/lflang/diagram/synthesis/util/ReactorIcons.java [hunk 3]
```diff
@@ -106,7 +106,7 @@ public void handleIcon(KContainerRendering rendering, ReactorDecl reactor, boole
         //                if (error != null) {
         //                    var errorText = _kContainerRenderingExtensions.addText(rendering,
         // "Icon not found!\n"+error);
-        //                    _kRenderingExtensions.setForeground(errorText, Colors.RED);
+        //                    _kRenderingExtensions.setForeground(errorText, Colors.BLUE);
```
Comment (lhstrh) [inline core/src/main/java/org/lflang/diagram/synthesis/util/ReactorIcons.java:109]: Seems like an accidental change?

Comment (edwardalee) [reply core/src/main/java/org/lflang/diagram/synthesis/util/ReactorIcons.java:109]: Oops. Reversed with 938bed159.

File: test/C/src/concurrent/WatchdogAction.lf (hunk 2 above)
Comment (edwardalee) [reply test/C/src/concurrent/WatchdogAction.lf]: Replaced with 
```
  // Offset may reduce the likelihood of flakiness if long startup times occur.
```

**Review by edwardalee — APPROVED** [review]
I think this is ready to merge.  Watchdogs still do not work with modes, but I've opened issue #2360.

**Pull request merged**